File: MPI/mpicheck/tests/functions/4.0/MPI_Register_datarep_c.f

        
subroutine var_1(val, ierr)
    integer val, ierr
end subroutine

subroutine var_2(val, ierr)
    integer val, ierr
end subroutine

subroutine var_3(val, ierr)
    integer val, ierr
end subroutine
        program main
        include 'mpif.h'
        
        CHARACTER*(10) var_0
       EXTERNAL var_1
       EXTERNAL var_2
       EXTERNAL var_3
       INTEGER(KIND=MPI_ADDRESS_KIND) var_4
       INTEGER var_5
        call mpi_register_datarep_c(var_0, var_1, var_2, var_3, var_4, var_5)
       call pmpi_register_datarep_c(var_0, var_1, var_2, var_3, var_4, var_5)
        end program main
    

File: MPI/mpicheck/tests/functions/4.0/MPI_Register_datarep_c.c
#include <mpi.h>
int main(char argc, char**argv)
{
    /* vars */
    const char *var_0;
    MPI_Datarep_conversion_function_c *var_1;
    MPI_Datarep_conversion_function_c *var_2;
    MPI_Datarep_extent_function *var_3;
    void *var_4;
    int ret;
    /* calls */
    ret = MPI_Register_datarep_c(var_0, var_1, var_2, var_3, var_4);
    ret = PMPI_Register_datarep_c(var_0, var_1, var_2, var_3, var_4);
    return 0;
}
